common/src/test.rs

```diff
@@ -0,0 +1,183 @@
+                            }
+                            Ok(())
+                        })
+                    })
+                    .with_filter(filter())
+            }))
+            .init();
+    });
+}
+
+/// A simple test logger that defaults to the INFO level
+#[cfg(all(target_family = "wasm", target_os = "unknown"))]
+pub fn logger() {
+    use tracing_subscriber::layer::SubscriberExt;
+    use tracing_subscriber::util::SubscriberInitExt;
+    use tracing_subscriber::EnvFilter;
+
+    INIT.get_or_init(|| {
+        let filter = EnvFilter::builder()
+            .with_default_directive(tracing::metadata::LevelFilter::DEBUG.into())
+            .from_env_lossy();
+
+        tracing_subscriber::registry()
+            .with(tracing_wasm::WASMLayer::default())
+            .with(filter)
+            .init();
+
+        console_error_panic_hook::set_once();
+    });
+}
+
+pub fn rand_hexstring() -> String {
+    let mut rng = crypto_utils::rng();
+    let hex_chars = "0123456789abcdef";
+    let v: String = (0..40)
+        .map(|_| hex_chars.chars().choose(&mut rng).unwrap())
+        .collect();
+
+    format!("0x{}", v)
+}
+
+pub fn rand_account_address() -> String {
+    Alphanumeric.sample_string(&mut crypto_utils::rng(), 42)
+}
+
+pub fn rand_vec<const N: usize>() -> Vec<u8> {
+    crate::rand_array::<N>().to_vec()
+}
+
+pub fn rand_u64() -> u64 {
+    crypto_utils::rng().gen()
+}
+
+#[cfg(not(target_arch = "wasm32"))]
+pub fn tmp_path() -> String {
+    let db_name = crate::rand_string::<24>();
+    format!("{}/{}.db3", std::env::temp_dir().to_str().unwrap(), db_name)
+}
+
+#[cfg(target_arch = "wasm32")]
+pub fn tmp_path() -> String {
+    let db_name = crate::rand_string::<24>();
+    format!("{}/{}.db3", "test_db", db_name)
+}
+
+pub fn rand_time() -> i64 {
+    let mut rng = rand::thread_rng();
+    rng.gen_range(0..1_000_000_000)
+}
+
+pub async fn wait_for_some<F, Fut, T>(f: F) -> Option<T>
+where
+    F: Fn() -> Fut,
+    Fut: Future<Output = Option<T>>,
+{
+    crate::time::timeout(crate::time::Duration::from_secs(60), async {
+        loop {
+            if let Some(r) = f().await {
+                return r;
+            } else {
+                crate::yield_().await;
+            }
+        }
+    })
+    .await
+    .ok()
+}
+
+pub async fn wait_for_ok<F, Fut, T, E>(f: F) -> Result<T, Expired>
+where
+    F: Fn() -> Fut,
+    Fut: Future<Output = Result<T, E>>,
+{
+    crate::time::timeout(crate::time::Duration::from_secs(60), async {
+        loop {
+            if let Ok(r) = f().await {
+                return r;
+            } else {
+                crate::yield_().await;
+            }
+        }
+    })
+    .await
+}
+
+pub async fn wait_for_eq<F, Fut, T>(f: F, expected: T) -> Result<(), Expired>
+where
+    F: Fn() -> Fut,
+    Fut: Future<Output = T>,
+    T: std::fmt::Debug + PartialEq,
+{
+    let result = crate::time::timeout(crate::time::Duration::from_secs(60), async {
```

Comment:
Can we reduce the default timeout to something like 2-3 seconds?

Reply:
I can reduce it a bit, but reducing too much causes tests in webassembly to flake out. I figure the common case of test success it will still finish quickly, but we need to give liberal time for wasm. I'll lower it to ~20sec?